services/node_util.go

```diff
@@ -79,25 +80,35 @@ func getNodeListForUpgrade(kubeClient *kubernetes.Clientset, hostsFailed *sync.M
 		if inactiveHosts[node.Labels[k8s.HostnameLabel]] {
 			continue
 		}
-		if val, ok := node.Labels[k8s.IgnoreHostDuringUpgradeLabel]; ok && val == "true" {
+		if val, ok := node.Labels[k8s.IgnoreHostDuringUpgradeLabel]; ok && val == k8s.IgnoreLabelValue {
 			continue
 		}
 		nodeList = append(nodeList, node)
 	}
 	return nodeList, nil
 }
 
-func CalculateMaxUnavailable(maxUnavailableVal string, numHosts int) (int, error) {
+func CalculateMaxUnavailable(maxUnavailableVal string, numHosts int, role string) (int, error) {
 	// if maxUnavailable is given in percent, round down
 	maxUnavailableParsed := k8sutil.Parse(maxUnavailableVal)
 	logrus.Debugf("Provided value for maxUnavailable: %v", maxUnavailableParsed)
+	if maxUnavailableParsed.Type == k8sutil.Int {
+		if maxUnavailableParsed.IntVal <= 0 {
+			return 0, fmt.Errorf("invalid input for max_unavailable_%s: %v, value must be > 0", role, maxUnavailableParsed.IntVal)
+		}
+	} else {
+		if strings.HasPrefix(maxUnavailableParsed.StrVal, "-") && maxUnavailableParsed.StrVal == "0%" {
+			return 0, fmt.Errorf("invalid input for max_unavailable_%s: %v, percentage input should be > 0", role, maxUnavailableParsed.StrVal)
```

Comment:
Small Nit: But in theory, it should have the same message as above to say `value must be > 0`